earth2studio/statistics/acc.py

```diff
@@ -0,0 +1,137 @@
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import torch
+
+from earth2studio.data import DataSource
+from earth2studio.data.utils import fetch_data
+from earth2studio.statistics.utils import _broadcast_weights
+from earth2studio.utils.type import CoordSystem
+
+
+class acc:
+    """
+    Statistic for calculating the anomaly correlation coefficient of two tensors
+    over a set of given dimensions, with respect to some optional climatology.
+
+    Parameters
+    ----------
+    reduction_dimensions: List[str]
+        A list of names corresponding to dimensions to perform the
+        statistical reduction over. Example: ['lat', 'lon']
+    climatology: DataSource
+        Optional (by default None) climatology to remove from tensors to create
+        anomalies before computing the correlation coefficient.
+    weights: torch.Tensor = None
+        A tensor containing weights to assign to the reduction dimensions.
+        Note that these weights must have the same number of dimensions
+        as passed in reduction_dimensions.
+        Example: if reduction_dimensions = ['lat', 'lon'] then
+        assert weights.ndim == 2.
+    """
+
+    def __init__(
+        self,
+        reduction_dimensions: list[str],
+        climatology: DataSource = None,
+        weights: torch.Tensor = None,
+    ):
+
+        if weights is not None:
+            if weights.ndim != len(reduction_dimensions):
+                raise ValueError(
+                    "Error! Weights must be the same dimension as reduction_dimensions"
+                )
+        self.reduction_dimensions = reduction_dimensions
+        self.weights = weights
+
+        self.climatology = climatology
+
+    def __str__(self) -> str:
+        return "acc"
+
+    def __call__(
+        self,
+        x: torch.Tensor,
+        x_coords: CoordSystem,
+        y: torch.Tensor,
+        y_coords: CoordSystem,
+    ) -> tuple[torch.Tensor, CoordSystem]:
+        """
+        Apply metric to data `x` and `y`, checking that their coordinates
+        are broadcastable.
+
+        Parameters
+        ----------
+        x : torch.Tensor
+            Input tensor #1 intended to apply metric to.
+        x_coords : CoordSystem
+            Ordered dict representing coordinate system that describes the `x` tensor.
+            'reduction_dims' must be in coords.
+        y : torch.Tensor
+            Input tensor #2 intended to apply statistic to.
+        y_coords : CoordSystem
+            Ordered dict representing coordinate system that describes the `y` tensor.
+            'reduction_dims' must be in coords.
+
+        Returns
+        -------
+        tuple[torch.Tensor, CoordSystem]
+            Returns anomaly correlation coefficient tensor with appropriate reduced coordinates.
```

Comment:
Similar to Modulus consider adding:

 Note
 ----
 For additional information see reference: 
 
 - https://www.atmos.albany.edu/daes/atmclasses/atm401/spring_2016/ppts_pdfs/ECMWF_ACC_definition.pdf